pkg/k8s/watchers/cilium_node.go

```diff
@@ -129,27 +129,21 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient client.Clientset, asyncContro
 // from kube-apiserver.
 func (k *K8sWatcher) GetCiliumNode(ctx context.Context, nodeName string) (*cilium_v2.CiliumNode, error) {
 	var (
-		err                      error
-		nodeInterface            interface{}
-		exists, getFromAPIServer bool
+		err           error
+		nodeInterface interface{}
+		exists        bool
 	)
 	k.ciliumNodeStoreMU.RLock()
 	// k.ciliumNodeStore might not be set in all invocations of GetCiliumNode,
-	// for example, during Cilium initialization GetCiliumNode is called from
-	// WaitForNodeInformation, which happens before ciliumNodeStore,
-	// so we will fallback to perform an API request to kube-apiserver.
+	// for example, when the key-value store is connected and the local store might
+	// have stale versions of the CiliumNode resource.
 	if k.ciliumNodeStore == nil {
-		getFromAPIServer = true
+		return k.clientset.CiliumV2().CiliumNodes().Get(ctx, nodeName, v1.GetOptions{})
 	} else {
 		nodeInterface, exists, err = k.ciliumNodeStore.GetByKey(nodeName)
```

Comment:
Why aren't we still failing back to the kubeapiserver in case it doesn't exist on the local store?

Reply:
My understanding is that the fallback was needed for the Egress Gateway manager, but now that it uses Resource[CiliumNode] it should not be needed anymore (see [this](https://github.com/cilium/cilium/pull/29222/commits/24736e1bc9767a7279667806f065a23c817a9f30) commit message for the references)

Reply:
So, looking again at the commit, I'm afraid I was confused with a related previous change to the CEP watcher. I'll remove this commit, but I'll keep this behavior (no fallback to the k8s apiserver when local store is not nil) in the last commit. That's because in the last commit we move the watcher to rely on Resource[T] and Resource[T] blocks until Store is fully synchronized: https://github.com/cilium/cilium/blob/fa0037638534635471c8e45583d78f87ad78b3e1/pkg/k8s/resource/resource.go#L262-L269